nrtk_explorer/app/core.py

```diff
@@ -41,73 +42,53 @@ def image_id_to_result(image_id):
 # ---------------------------------------------------------
 
 
-class Engine:
+class Engine(Applet):
     def __init__(self, server=None):
         if server is None:
             server = get_server()
 
-        self._server = server
+        super().__init__(server)
 
-        self.server._local_state = {
-            "image_objects": {},
-        }
+        self.context["image_objects"] = {}
+        self.context["images_manager"] = images_manager.ImagesManager()
+        self.context["annotations"] = {}
```

Comment:
you can use the `.` notation too.

Reply:
Yes, it's a leftover from the fact we were using a dict for the `local_state`.
So is the `.` notation preferred over `[]` when using a trame state object?

Reply:
yes, it is a more natural way unless you want to do something like

```
for item in items:
   self.context[item] = ...
```